### PR TITLE
Add configurable timeout to HTTP connections

### PR DESCRIPTION
**Pull request by SzumiecM**
## Proposed Changes

I added a configurable timeout which solves HTTP connections hanging on occasions of the server being overloaded.

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes issue #NNNN)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [ ] Cosmetic change (whitespace, formatting, etc)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask on the
mailing list. We're here to help! This is simply a reminder of what we are
going to look for before merging your code._

- [x] I have read the `CONTRIBUTING.md` document
- [ ] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [ ] All tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in related repositories

## Further Comments

If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc.

### CODE REVIEW

**Comment by michaelklishin** [comment]
Thank you. These new settings refer to the `httpc` settings but I find them to be pretty terrible. It's not clear what the difference is. What is a `timeout`? Why is there a `connect_timeout`? (a grammatically poor name, too)

File: deps/rabbitmq_auth_backend_http/examples/rabbitmq_auth_backend_spring_boot_kotlin/src/main/resources/rabbitmq.conf [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 # http backend
 auth_backends.1 = rabbit_auth_backend_http
+auth_http.timeout     = 10000
```
Comment (michaelklishin) [inline deps/rabbitmq_auth_backend_http/examples/rabbitmq_auth_backend_spring_boot_kotlin/src/main/resources/rabbitmq.conf:7]: Can we rename this to `auth_http.request_timeout`?

File: deps/rabbitmq_auth_backend_http/examples/rabbitmq_auth_backend_spring_boot_kotlin/src/main/resources/rabbitmq.conf [hunk 2]
```diff
@@ -4,6 +4,8 @@
...
+auth_http.connect_timeout     = 5000
```
Comment (michaelklishin) [inline deps/rabbitmq_auth_backend_http/examples/rabbitmq_auth_backend_spring_boot_kotlin/src/main/resources/rabbitmq.conf:8]: Let's rename this to `auth_http.connection_timeout`. In fact, that's what [`httpc`'s own doc reference](http://erlang.org/doc/man/httpc.html#set_options-1) calls this timeout :/

**Review by michaelklishin — CHANGES_REQUESTED** [review]

**Comment by michaelklishin** [comment]
@SzumiecM I will do the renaming on your behalf and hope to get it into `3.8.18`.

**Comment by SzumiecM** [comment]
Thank you for your approval, and also for all the comments and help.

**Comment by michaelklishin** [comment]
Backported to `v3.8.x`.

**Pull request merged**